graphbolt/CMakeLists.txt

```diff
@@ -87,6 +107,11 @@ if(USE_CUDA)
   
   get_property(archs TARGET ${LIB_GRAPHBOLT_NAME} PROPERTY CUDA_ARCHITECTURES)
   message(STATUS "CUDA_ARCHITECTURES for graphbolt: ${archs}")
+
+  get_property(archs TARGET ${LIB_GRAPHBOLT_CUDA_NAME} PROPERTY CUDA_ARCHITECTURES)
+  message(STATUS "CUDA_ARCHITECTURES for graphbolt extension: ${archs}")
```

Comment:
@Rhett-Ying the output in the CI is:
```
-- CUDA_ARCHITECTURES for graphbolt: 35;50;60;70;75;80;86;90
-- CUDA_ARCHITECTURES for graphbolt extension: 70;75;80;86;90
```

However, it looks like the extension is being compiled for more CUDA architectures than it is being reported. Do you know why that could be?

Reply:
`LIB_GRAPHBOLT_NAME` is applied already for some libraries? why do we need a new one named `LIB_GRAPHBOLT_CUDA_NAME` instead of filter archs from existing one?

Reply:
Because we compile all of graphbolt except `cuda/extension` folder with unfiltered CUDA architectures. Only `cuda/extension` folder gets compiled with filtered architectures. I figured out that TORCH_CUDA_ARCH_LIST interferes with the architecture list so I am hardcoding it to Volta as we receive the architecture list from DGL.

Reply:
To be able to specify different architecture lists, I needed to create a separate library called `LIB_GRAPHBOLT_CUDA_NAME` and linked it statically to `LIB_GRAPHBOLT_NAME`.

Reply:
https://github.com/NVIDIA/cuCollections/issues/449 for the workaround.

Reply:
why is `volta`?

Reply:
It is exactly compute capability 70. If `TORCH_CUDA_ARCH_LIST` is left empty, torch injects to nvcc all architectures, including lower than 70. In that case, my extension also gets compiled with architectures that are lower than 70 and fails compilation. This is a workaround until the CCCL library fixes the CUDA architecture issue of the features they have in their library.

Setting TORCH_CUDA_ARCH_LIST to Volta makes it so that only 70 architecture is injected into nvcc and my extension is okay with that.

Reply:
This workaround can be removed when https://github.com/NVIDIA/cccl/issues/1083 is resolved.

Reply:
what if anyone build DGL from source on Pascal if hardcode with `volta`?

Reply:
They can still do that by passing -DCUDA_ARCH_NAME=Pascal to DGL's cmake argument.

Reply:
Or if they don't pass anything, their GPU will be auto recognized and it will work as normal.

Reply:
But I'm afraid the error message may not be intuitive as they never expect we hardcoded `volta` in build script.

Reply:
The TORCH_CUDA_ARCH_LIST is parallel to what is coming from DGL. So having it hardcoded only makes it so that they will build for Pascal and Volta.

Reply:
> But I'm afraid the error message may not be intuitive as they never expect we hardcoded `volta` in build script.

What could cause an error here?

Reply:
If this is the best workaround for now, please add the detailed explanation alongside the hardcode. and paste the dependent https://github.com/NVIDIA/cccl/issues/1083 as well

Reply:
Yeah, I spent days over this. I feel this is the best I have got unfortunately. If torch allowed us to pass no architectures (and injected no cuda architectures into nvcc when that is the case), we wouldn't need to hard code Volta there. If you find a better solution, please ping me.

Reply:
> > But I'm afraid the error message may not be intuitive as they never expect we hardcoded `volta` in build script.
> 
> What could cause an error here?

if user build without specifying `-DCUDA_ARCH_NAME=Pascal`, will build fail?

Reply:
If they don't specify anything but have a GPU in their system, it will build GraphBolt for Pascal and Volta.

Reply:
And Build DGL for Pascal.

Reply:
got it.